tests/Doctrine/Tests/ORM/Tools/SchemaValidatorTest.php

```diff
@@ -675,3 +700,31 @@ abstract class Issue9095AbstractChild extends Issue9095Parent
 class Issue9095Child extends Issue9095AbstractChild
 {
 }
+
+/** @Entity */
+class InvalidEntity3
+{
+    /**
+     * @var int
+     * @Id
+     * @Column
+     */
+    protected $key;
+
+    /**
+     * @Column(type="boolean")
+     */
+    protected int $property;
```

Comment:
Ooooops, the test fails here because Doctrine didn't yet drop PHP 7.3 support (REALLY?!)

I think the best way to handle this would be to:

 1. split this out to a separate file
 2. mark the tests with `@requires php>=7.4`
 3. exclude the separated file from analysis by tooling like phpcs/phpstan

Reply:
How would you prevent the file from being read with php < 7.4?
The test should not be executed with a `@requires php 70400` but the file contains some classes definition too...

Reply:
I don't remember how the test suite works precisely, but can check it tomorrow.

I'd say something like `require_once __DIR__ . '/some-php-7.4-only-file.php` and then reference the classes in that file that way

Reply:
Or use autoload, or team up with @MatTheCat who is having the same issue on https://github.com/doctrine/orm/pull/10666/files#r1186698321

We could have a directory with PHP8-specific tests, and then future contributors would not need to bother with `@requires` or with splitting test files and companion classes.